test/provenance.cpp

```diff
@@ -0,0 +1,244 @@
+//*****************************************************************************
+// Copyright 2017-2019 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+
+#include <fstream>
+#include <sstream>
+#include <string>
+#include <vector>
+
+#include "gmock/gmock.h"
+#include "gtest/gtest.h"
+
+#include "ngraph/ngraph.hpp"
+#include "ngraph/provenance.hpp"
+
+using namespace std;
+using namespace ngraph;
+using ::testing::Return;
+
+using ProvSet = std::unordered_set<std::string>;
+
+TEST(provenance, provenance)
+{
+    set_provenance_enable(true);
```

Comment:
It's not likely to matter, but let's store the current value from `get_provenance_enable()` and restore it at the bottom of the test.

Reply:
Or RAII it.